python/foglamp/services/common/microservice.py

```diff
@@ -92,12 +92,19 @@ def _get_service_registration_payload(self):
             }
         return service_registration_payload
 
+    async def shutdown(self, request=None):
+        await self.stop()
+        self.unregister_service_with_core(self._microservice_id)
+        return web.json_response({"message":
+            "Successfully shutdown microservice id {} at url http://{}:{}/foglamp/service/shutdown".format(
+            self._microservice_id, self._microservice_management_host, self._microservice_management_port)})
+
     @abstractmethod
-    async def shutdown(self):
+    async def stop(self, request=None):
```

Comment:
I don't think we should change it to stop!

Reply:
it will be implemented by  microservice itself; and will hadnle request from microservice managment client api; However it will be  interesting to see how shall we stop aiohttp instance that is allowing this route endpoint  itself